_posts/2018-02-06-unreliable-network.md

```diff
@@ -0,0 +1,116 @@
+---
+layout: post
+title: Quick Tips for App Developers on Surviving with Unreliable Network
+author: jarzka
+excerpt: Methods that you, as an application developer, can do to make your app feel more stable even if the underlying network is unreliable.
+tags:
+- clojure
+- clojurescript
+- network programming
+---
+
+You cannot always choose where your application will be hosted. Sometimes the network infrastructure can be unreliable, meaning that network requests can at times take a long time to complete and requests will fail at random rates. Even if these problems could not be prevented from happening, luckily we, as application developers, have some tricks we can use to make our applications *feel* more stable in unreliable network. And even if your network infrastucture is solid, it is not guaranteed that occasional communication failures wonwill not happen, as illustrated in [Two Generals Problem](https://en.wikipedia.org/wiki/Two_Generals%27_Problem). Thus, these quick tips, targeted mainly at single page web application and mobile application developers, will probably help your application appear more stable and responsive for your end users.
```

Comment:
~could not be prevented from happening~ can't be prevented, …

~in unreliable network~ over an unreliable network

~won~will

as illustrated in *the* Two Generals Problem.

(Also *single-page*, but not really a big deal.)